rootex/core/renderer/shaders/decal_pixel_shader.hlsl

```diff
@@ -0,0 +1,104 @@
+#include "register_locations_pixel_shader.h"
+#include "basic_material.hlsli"
+
+Texture2D ShaderTexture : register(DIFFUSE_PS_HLSL);
+Texture2D NormalTexture : register(NORMAL_PS_HLSL);
+Texture2D SpecularTexture : register(SPECULAR_PS_HLSL);
+Texture2D LightmapTexture : register(LIGHTMAP_PS_HLSL);
+Texture2D DepthTexture : register(DEPTH_TEXTURE_PS_HLSL);
+SamplerState SampleType : register(SAMPLER_PS_HLSL);
+
+struct DecalPixelInputType
+{
+	float4 screenPosition : SV_POSITION;
+	float3 normal : NORMAL;
+	float4 worldPosition : POSITION;
+	float2 tex : TEXCOORD0;
+	float3 tangent : TANGENT;
+	float4 color : COLOR;
+	nointerpolation float3 decalRight : TEXCOORD1;
+	nointerpolation float3 decalForward : TEXCOORD2;
+	nointerpolation float3 decalUp : TEXCOORD3;
+	nointerpolation float3 decalHalfScale : TEXCOORD4;
+	nointerpolation float4 decalViewspacePosition : TEXCOORD5;
```

Comment:
can this be shifted to a constant buffer? 
in case not, change semantics, TEXCOORD* imply UV coords